### PR TITLE
feat!: Enable autoscaling by default

### PR DESCRIPTION
**Pull request by dmolenda-sumo**

### CODE REVIEW

File: deploy/helm/sumologic/values.yaml [hunk 1]
```diff
@@ -454,7 +458,7 @@ sumologic:
         ## Option to turn autoscaling on for otelcol and specify params for HPA.
         ## Autoscaling needs metrics-server to access cpu metrics.
         autoscaling:
-          enabled: false
+          # enabled: true
```
Comment (sumo-drosiek) [inline deploy/helm/sumologic/values.yaml:461]: shouldn't be `false` as well?

Comment (swiatekm-sumo) [reply deploy/helm/sumologic/values.yaml:461]: To the contrary, shouldn't it be true everywhere? That's the actual default after this change.

Comment (sumo-drosiek) [reply deploy/helm/sumologic/values.yaml]: What's the point of uncommenting it if it's true anyway?
We are using `#` more like examples in other cases in this file

**Comment by swiatekm-sumo** [comment]
Looks good to me, overall. In general, the `basic.input.yaml` files shouldn't have any changes - we should see changes in the output, because autoscaling is now enabled. We should disable autoscaling in `custom.input.yaml` golden files, because those files exists to check the effects of changing defaults. Right now, you have a mix of these.

File: deploy/helm/sumologic/templates/_helpers/_common.tpl [hunk 2]
```diff
@@ -527,3 +527,19 @@ podAntiAffinity:
...
 {{- end -}}
+
+{{/*
+Returns whether autoscaling is enabled. 
+Returns value of received parameter (.autoscalingEnabled) if defined, 
+otherwise returns value of global flag (.Values.sumologic.autoscaling.enabled).
+
+Example usage:
+{{- template "is.autoscaling.enabled" (dict "autoscalingEnabled" .Values.metadata.logs.autoscaling.enabled) }}
+*/}}
```
Comment (swiatekm-sumo) [inline deploy/helm/sumologic/templates/_helpers/_common.tpl:538]: Looking at actual usage examples, this isn't correct. I think it should be
```suggestion
Example usage:
{{- template "is.autoscaling.enabled" (dict "autoscalingEnabled" .Values.metadata.logs.autoscaling.enabled "Values" .Values) }}
*/}}
```

**Review by swiatekm-sumo — APPROVED** [review]
LGTM

**Pull request merged**